### ISSUE TITLE
Sudden, often, and random crash (application quits)

### ISSUE THREAD

**Issue by unlucio** [brave/brave-browser#17054]
!!! PLEASE, I know this is probably the worst bug report ever as I don't even have a consistent way to reproduce the problem.
But that's pretty much the problem its self: for some reason today Brave started randomly crashing (a lot) and I cna't find anything giving me a clue on what/why!!!!

## Description 
As I'm browsing, Brave suddenly closes. No hanging, no "send to apple" crash report. Just the app suddenly vanishes and closes its self.
If I open it again and click on the "restore tabs button" it crashes again and again until I just give up and start fresh.
This has been happening for the best part of this afternoon/evening (~5h).
It started randomly and out of the blue. 1st on youtube, but not only limited to that.

## Steps to Reproduce
!!! Please understand that due to the randomness and lack of any feedback I'm experiencing, these steps will look silly to those reading!!!
   1. Open Brave
   2. Navigate
   3. At some point it will crash
   4. Open it again
   5. Restore Your tabs and watch Brave crash and vanish before your eyes
   6. Open, restore, and witness it go again and again.

## Actual result:
All the windows are gone, application closed.

## Expected result:
well, to keep browsing ;P

## Reproduces how often: 
It started today around 14CET, random events but often enough to be noticeable and now annoying enough o warrant a report.

## Brave version (brave://version info)
Brave	1.26.77 Chromium: 91.0.4472.164 (Official Build) (x86_64)
Revision	541163496c9982c98f61819bab7cf2183ea8180f-refs/branch-heads/4472@{#1569}
OS	macOS Version 11.4 (Build 20F71)
JavaScript	V8 9.1.269.39
User agent	Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.164 Safari/537.36
Command Line	./Brave Browser --enable-dom-distiller --disable-domain-reliability --no-pings --origin-trial-public-key=bYUKPJoPnCxeNvu72j4EmPuK7tr1PAC7SHh8ld9Mw3E=,fMS4mpO6buLQ/QMd+zJmxzty/VQ6B1EUZqoCU04zoRU= --sync-url=https://sync-v2.brave.com/v2 --lso-url=https://no-thanks.invalid --variations-server-url=https://variations.brave.com/seed --enable-features=PrefetchPrivacyChanges,PasswordImport,ReducedReferrerGranularity,SafetyTip,AutoupgradeMixedContent,WebUIDarkMode,LegacyTLSEnforced --disable-features=AutofillServerCommunication,EnableProfilePickerOnStartup,TextFragmentAnchor,AutofillEnableAccountWalletStorage,FledgeInterestGroupAPI,HandwritingRecognitionWebPlatformApiFinch,FlocIdComputedEventLogging,HandwritingRecognitionWebPlatformApi,InterestCohortFeaturePolicy,FledgeInterestGroups,SubresourceWebBundles,NetworkTimeServiceQuerying,LangClientHintHeader,NotificationTriggers,WebOTP,FirstPartySets,InterestCohortAPIOriginTrial,TrustTokens,DirectSockets,FederatedLearningOfCohorts,LiveCaption,IdleDetection,SignedExchangePrefetchCacheForNavigations,SignedExchangeSubresourcePrefetch --flag-switches-begin --enable-experimental-web-platform-features --flag-switches-end
Executable Path	/Applications/Brave Browser.app/Contents/MacOS/Brave Browser
Profile Path	/Users/lucio/Library/Application Support/BraveSoftware/Brave-Browser/Default
Variations	AdRewardsStudy:NextPaymentDay
EphemeralStorageStudy:Enabled
NativeCosmeticFilteringStudy:Enabled
PermissionLifetimeReleaseStudy:Enabled

OS: macOS Big Sur 11.4 (20F71)

## Version/Channel Information:
I just use the official release.

## Other Additional Information:
- Does the issue resolve itself when disabling Brave Shields?
Hard to tell: as it happens randomly I'm not really willing to browse around with the shield down, as that's the reason I use Brave ^^'
- Does the issue resolve itself when disabling Brave Rewards?
Again I haven't had the opportunity to check this
- Is the issue reproducible on the latest version of Chrome?
No idea, I'm not super-keen about having google's vacuums running on my machine.

## Miscellaneous Information:
I understand this is a very random thing. What I'm seeking right now is:
- am I the only one suddenly experiencing this issue (mostly on youtube, but not only)
- what can I do to produce traces/logs useful for debugging this issue?

I already tried clearing caches and everything cleanable but it didn't help. I tried everything that makes sense as a "user knowing his computer", but without wearing my "software engineer hat" (sorry I don't have time right now to debug this, but I'm happy to gather logs and traces if they're convenient to find/produce)

**Comment by rebron** [comment]
@unlucio When you launch brave next time can you go to brave://crashes/ send the report and copy and paste the crash ID here so we can take a look?

**Comment by unlucio** [comment]
With a bit of luck, I was able to capture this output by launching brave from the shell:
```sh
➜  MacOS ./Brave\ Browser
[89327:259:0720/192644.475904:ERROR:profile_attributes_storage.cc(547)] Failed to PNG encode the image.
Failed to parse JSON adblock resources: EOF while parsing a value at line 1 column 0
[89327:259:0720/192647.278968:ERROR:CONSOLE(1)] "[Shields]: Can't request shields panel data. Error: No tab url specified", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/192647.279124:ERROR:CONSOLE(1)] "[Shields]: Can't request shields panel data. Error: No tab url specified", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/193710.044929:ERROR:CONSOLE(1)] "[Shields]: Can't request shields panel data. Error: No tab url specified", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/193710.047842:ERROR:CONSOLE(1)] "[Shields]: Can't request shields panel data. Error: No tab url specified", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/193713.729177:ERROR:CONSOLE(1)] "[Shields]: Can't request shields panel data. Error: No tab url specified", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/193713.729313:ERROR:CONSOLE(1)] "[Shields]: Can't request shields panel data. Error: No tab url specified", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/194523.111530:ERROR:CONSOLE(1)] "Failed to handle media duration metadata: missing publisher key for media key youtube_fqpSbdkjBEg", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/194523.112121:ERROR:CONSOLE(1)] "Failed to handle media duration metadata: missing publisher key for media key youtube_fqpSbdkjBEg", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/194533.094512:ERROR:CONSOLE(1)] "Failed to handle media duration metadata: missing publisher key for media key youtube_fqpSbdkjBEg", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/194533.094727:ERROR:CONSOLE(1)] "Failed to handle media duration metadata: missing publisher key for media key youtube_fqpSbdkjBEg", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/194543.385010:ERROR:CONSOLE(1)] "Failed to handle media duration metadata: missing publisher key for media key youtube_fqpSbdkjBEg", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/194543.385445:ERROR:CONSOLE(1)] "Failed to handle media duration metadata: missing publisher key for media key youtube_fqpSbdkjBEg", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/194606.101897:ERROR:CONSOLE(1)] "Failed to handle media duration metadata: missing publisher key for media key youtube_fqpSbdkjBEg", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/194606.102658:ERROR:CONSOLE(1)] "Failed to handle media duration metadata: missing publisher key for media key youtube_fqpSbdkjBEg", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/194646.100140:ERROR:CONSOLE(1)] "Failed to handle media duration metadata: missing publisher key for media key youtube_fqpSbdkjBEg", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/194646.100608:ERROR:CONSOLE(1)] "Failed to handle media duration metadata: missing publisher key for media key youtube_fqpSbdkjBEg", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/194726.385082:ERROR:CONSOLE(1)] "Failed to handle media duration metadata: missing publisher key for media key youtube_fqpSbdkjBEg", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/194726.385485:ERROR:CONSOLE(1)] "Failed to handle media duration metadata: missing publisher key for media key youtube_fqpSbdkjBEg", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/194809.156681:ERROR:CONSOLE(1)] "Failed to handle media duration metadata: missing publisher key for media key youtube_fqpSbdkjBEg", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/194809.157008:ERROR:CONSOLE(1)] "Failed to handle media duration metadata: missing publisher key for media key youtube_fqpSbdkjBEg", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/194846.119288:ERROR:CONSOLE(1)] "Failed to handle media duration metadata: missing publisher key for media key youtube_fqpSbdkjBEg", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/194846.119584:ERROR:CONSOLE(1)] "Failed to handle media duration metadata: missing publisher key for media key youtube_fqpSbdkjBEg", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/194926.317460:ERROR:CONSOLE(1)] "Failed to handle media duration metadata: missing publisher key for media key youtube_fqpSbdkjBEg", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/194926.317817:ERROR:CONSOLE(1)] "Failed to handle media duration metadata: missing publisher key for media key youtube_fqpSbdkjBEg", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/195006.103966:ERROR:CONSOLE(1)] "Failed to handle media duration metadata: missing publisher key for media key youtube_fqpSbdkjBEg", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/195006.105173:ERROR:CONSOLE(1)] "Failed to handle media duration metadata: missing publisher key for media key youtube_fqpSbdkjBEg", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/195046.108140:ERROR:CONSOLE(1)] "Failed to handle media duration metadata: missing publisher key for media key youtube_fqpSbdkjBEg", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[89327:259:0720/195046.108780:ERROR:CONSOLE(1)] "Failed to handle media duration metadata: missing publisher key for media key youtube_fqpSbdkjBEg", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[0720/195456.626763:WARNING:crash_report_exception_handler.cc(240)] UniversalExceptionRaise: (os/kern) failure (5)
[1]    89327 segmentation fault  ./Brave\ Browser
```

and yup, seems like something in the shield goes bonkers

**Comment by rebron** [comment]
cc: @iefremov

**Comment by unlucio** [comment]
@rebron sorry I didn't see your answer.
Here's what's under my `brave://crashes/` page

<img width="1686" alt="Screenshot 2021-07-20 at 20 06 59" src="https://user-images.githubusercontent.com/778799/126373732-b55a8a5f-b5c8-48b3-81ca-4e5abdd25cde.png">


It says "Crash reporting is disabled.", yet I've "send now" button.
Shall I press all the "send now" buttons?

**Comment by keur** [comment]
@unlucio If you could also enable crash reporting in `brave://settings` we can get a trace. If you search `Help improve Brave's features and performance` in settings it should pop up

**Comment by unlucio** [comment]
ok, I activated the telemetry.
Now I just gotta wait and hope for a crash, weirdly 😆

**Comment by unlucio** [comment]
@keur here are a couple of IDS:
d2e52200-ad33-a506-0000-000000000000
3f4b2300-ad33-a506-0000-000000000000

for the crash

**Comment by iefremov** [comment]
Many thanks @unlucio, could you please share all of the IDS?

**Comment by iefremov** [comment]
@unlucio most probably this is caused by `--enable-experimental-web-platform-features` flag. Can you please open brave://flags and disable `enable-experimental-web-platform-features` ?

**Comment by unlucio** [comment]
@iefremov having only recently turned telemetry on I just have one more for you: 
`81982300-ad33-a506-0000-000000000000`

<img width="567" alt="Screenshot 2021-07-21 at 10 04 31" src="https://user-images.githubusercontent.com/778799/126453934-48ba8e93-2105-4595-8113-75e7ccb7f2e4.png">
tho I do have 3 other sessions not linked to a `reportID` for some reason

**Comment by iefremov** [comment]
Thanks @unlucio . So I mentioned above, disabling `enable-experimental-web-platform-feature` on brave://flags should fix the problem.

**Comment by unlucio** [comment]
@iefremov I disabled it now.

**Comment by Tonev** [comment]
@iefremov 

Community user was experiencing random crashes in YouTube. It turned out they had enabled the `enable-experimental-web-platform-features` flag so they could use the PiP feature.

Crashes ID's:
```
80c12600-ad33-a506-0000-000000000000
7ac12600-ad33-a506-0000-000000000000
76c12600-ad33-a506-0000-000000000000
```

Thought I would share this with you in case anything has to be done regarding the flag, which seems to cause crashes for some reason.

You can see the whole thread below:
https://community.brave.com/t/youtube-makes-brave-crash/268097

**Comment by iefremov** [comment]
Thanks @Tonev, those are exactly same crashes. I'm not sure anything can be done with the flag: this is basically the upstream code that crashes.
Also brave://flags contains pretty much visible caution saying that playing with experimental flags can make browser unstable. So people who change flags accept the risk.

**Comment by lightmyfire17** [comment]
Exactly the same YouTube crash issue.

![image](https://user-images.githubusercontent.com/19778699/126515150-858f86f0-9a02-4804-95da-678d5d7b9a1e.png)

**Comment by iefremov** [comment]
@lightmyfire17 and anyone else who experience this issue, please

1. Open brave://flags
2. Find enable-experimental-web-platform-feature and switch it to disabled state
3. Restart the browser

**Comment by unlucio** [comment]
Btw, I know this is slightly offtopic but:
I turned the experiments on in order to activate the support for the Bluetooth web API.
The BT API didn't activate and instead, I got this issue 😆
The community forums have some threads on BT but none of them were ever answered:how do you turn on BT on brave?

**Comment by iefremov** [comment]
@unlucio I think it is disabled, see https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)

**Comment by bridiver** [comment]
based on the PR there is no way to re-enable, but maybe we should use feature flags for that kind of thing so users can selectively re-enable at their own risk. cc @diracdeltas @pes10k

**Comment by unlucio** [comment]
That's a great idea @bridiver.
When it comes to JS/web APIs disabling them tout court might be slightly suboptimal: for instance, as an SW engineer, if I want to experiment with the BT web API right now I'm forced to download chrome, while I'd love to be able to do it on Brave with the peace of mind of a safer and more polite browser. 

Perhaps having a small explanation on why it's been disabled would help the user have an informed decision, I skimmed through the page kindly offered by @iefremov, and while it def explained why I want' able to use the webBT, I wasn't really able to grasp why it was removed. What potential problem does it constitute? Perhaps knowing it would be more interesting useful than not.
